src/journal/journal-verify.c

```diff
@@ -70,10 +70,11 @@ static void draw_progress(uint64_t p, usec_t *last_usec) {
 }
 
 static uint64_t scale_progress(uint64_t scale, uint64_t p, uint64_t m) {
+        /* Calculates scale * p / m, but handles m == 0 safely, and saturates.
+         * Currently all callers use m >= 1, but we keep the check to be defensive.
+         */
 
-        /* Calculates scale * p / m, but handles m == 0 safely, and saturates */
-
-        if (p >= m || m == 0)
+        if (p >= m || m == 0) // lgtm[cpp/constant-comparison]
```

Comment:
The original commit used the c-style comment, not the c++-style one. Even though it's really a triviality, I'd say that we should do the same here, as systemd uses the c-style comments everywhere.

Reply:
This is an unfortunate behavior of LGTM - it simply doesn't recognize suppression rules in C++-style comments. I had to do a similar change in upstream as well (see https://github.com/systemd/systemd/commit/1a862e21511918f4f7a27a119624e9629056356b) as the rule was ignored within a C++-style comment.

Excerpt from the [LGTM documentation](https://lgtm.com/help/lgtm/alert-suppression#c-c++):
```
The syntax is language-dependent:

    "// lgtm..." for C/C++, C#, Java and JavaScript
    "* lgtm..." for COBOL
    "# lgtm..." for Python
```